### DISCUSSION TITLE
error installing web3.py

### DISCUSSION THREAD

**Discussion by ZubairIsmail05** [smartcontractkit/full-blockchain-solidity-course-py#791] [General]
Hey I am new to coding and having trouble installing web3.py in Lesson 4. I am on mac if that is useful. The terminal and error looks as follows:
<img width="1130" alt="Screen Shot 2022-01-14 at 9 50 20 PM" src="https://user-images.githubusercontent.com/86219019/149606687-b72db332-5644-45a2-8050-d0fd0db42fd7.png">
<img width="1130" alt="Screen Shot 2022-01-14 at 9 50 50 PM" src="https://user-images.githubusercontent.com/86219019/149606688-2c58ab4e-eaa8-4e44-b8af-8ffc4beb8db6.png">
<img width="1123" alt="Screen Shot 2022-01-14 at 9 52 28 PM" src="https://user-images.githubusercontent.com/86219019/149606689-65728036-acda-4ff6-91dd-08ce7727c497.png">

**Comment by cromewar** [comment]
Hello @ZubairIsmail05 sorry for the delay.
This is a well known problem regarding to some libraries you'll need inf order to work with this, you must install:

- Xcode.
- Cython

Otherwise you won't be able to resolve this. Give it a try.
